File: Microsoft.Recognizers.Text.Number/Parsers/INumberParserConfiguration.cs
using System.Collections.Generic;
using System.Collections.Immutable;
using System.Globalization;
using System.Text.RegularExpressions;

namespace Microsoft.Recognizers.Text.Number.Parsers
{
    public interface INumberParserConfiguration
    {
        #region language dictionaries
        ImmutableDictionary<string, long> CardinalNumberMap { get; }

        ImmutableDictionary<string, long> OrdinalNumberMap { get; }

        ImmutableDictionary<string, long> RoundNumberMap { get; }
        #endregion

        #region language settings
        CultureInfo CultureInfo { get; }

        Regex DigitalNumberRegex { get; }

        string FractionMarkerToken { get; }

        Regex HalfADozenRegex { get; }

        string HalfADozenText { get; }

        string LangMarker { get; }

        char NonDecimalSeparatorChar { get; }

        char DecimalSeparatorChar { get; }

        string WordSeparatorToken { get; }

        IEnumerable<string> WrittenDecimalSeparatorTexts { get; }

        IEnumerable<string> WrittenGroupSeparatorTexts { get; }

        IEnumerable<string> WrittenIntegerSeparatorTexts { get; }

        IEnumerable<string> WrittenFractionSeparatorTexts { get; }
        #endregion

        /// <summary>
        /// Used when requiring to normalize a token to a valid expression supported by the ImmutableDictionaries (language dictionaries)
        /// </summary>
        /// <param name="tokens">list of tokens to normalize</param>
        /// <param name="context">context of the call</param>
        /// <returns>list of normalized tokens</returns>
        IEnumerable<string> NormalizeTokenSet(IEnumerable<string> tokens, object context);

        /// <summary>
        /// Used when requiring to convert a string to a valid number supported by the language
        /// </summary>
        /// <param name="numberStr">composite number</param>
        /// <returns>value of the string</returns>
        long ResolveCompositeNumber(string numberStr);
    }
}

File: Java/libraries/recognizers-text-number/src/main/java/com/microsoft/recognizers/text/number/parsers/INumberParserConfiguration.java
package com.microsoft.recognizers.text.number.parsers;

import com.microsoft.recognizers.text.CultureInfo;
import com.microsoft.recognizers.text.ParseResult;
import com.microsoft.recognizers.text.number.NumberOptions;

import java.util.List;
import java.util.Map;
import java.util.regex.Pattern;

public interface INumberParserConfiguration {

    //region language dictionaries

    Map<String, Long> getCardinalNumberMap();

    Map<String, Long> getOrdinalNumberMap();

    Map<String, Long> getRoundNumberMap();

    //endregion

    //region language settings

    NumberOptions getOptions();

    CultureInfo getCultureInfo();

    Pattern getDigitalNumberRegex();

    Pattern getFractionPrepositionRegex();

    String getFractionMarkerToken();

    Pattern getHalfADozenRegex();

    String getHalfADozenText();

    String getLangMarker();

    char getNonDecimalSeparatorChar();

    char getDecimalSeparatorChar();

    String getWordSeparatorToken();

    List<String> getWrittenDecimalSeparatorTexts();

    List<String> getWrittenGroupSeparatorTexts();

    List<String> getWrittenIntegerSeparatorTexts();

    List<String> getWrittenFractionSeparatorTexts();

    Pattern getNegativeNumberSignRegex();

    //endregion

    /**
     * Used when requiring to normalize a token to a valid expression supported by the ImmutableDictionaries (language dictionaries)
     *
     * @param tokens  list of tokens to normalize
     * @param context context of the call
     * @return list of normalized tokens
     */
    List<String> normalizeTokenSet(List<String> tokens, ParseResult context);

    /**
     * Used when requiring to convert a String to a valid number supported by the language
     *
     * @param numberStr composite number
     * @return value of the String
     */
    long resolveCompositeNumber(String numberStr);
}
